tests/lax_numpy_test.py

```diff
@@ -1179,6 +1179,25 @@ def np_fun(arg1, arg2):
       return np.union1d(arg1, arg2).astype(dtype)
     self._CheckAgainstNumpy(np_fun, jnp.union1d, args_maker)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_{}_{}_assume_unique={}".format(
+       jtu.format_shape_dtype_string(shape1, dtype1),
+       jtu.format_shape_dtype_string(shape2, dtype2),
+       assume_unique),
+       "shape1": shape1, "dtype1": dtype1, "shape2": shape2, "dtype2": dtype2,
+       "assume_unique": assume_unique}
+      for dtype1 in [s for s in default_dtypes if s != jnp.bfloat16]
+      for dtype2 in [s for s in default_dtypes if s != jnp.bfloat16]
+      for shape1 in one_dim_array_shapes
+      for shape2 in one_dim_array_shapes
+      for assume_unique in [False, True]))
```

Comment:
When `assume_unique` is True, we should make certain the inputs contain unique values, otherwise the output is ill-defined and it could cause non-meaningful test errors if either numpy or jax were to switch to a different implementation.

Reply:
Regarding this, I thought of using `rand_unique_int` instead of `rand_default` for rng when `assume_unique=True`   https://github.com/google/jax/blob/f63106c8485496750b9663b09813d11269c17c2a/jax/test_util.py#L718-L722
But using `rand_unique_int` will limit the test inputs to `int`s and also it throws exceptions when `shapes` is scalar due to the use of `prod(shapes)` in it. 

PS: Also while we're at it, this might be applicable to `testIntersect1d` as well.
https://github.com/google/jax/blob/f63106c8485496750b9663b09813d11269c17c2a/tests/lax_numpy_test.py#L1206-L1211